compiler_opt/rl/data_reader.py

```diff
@@ -193,9 +194,11 @@ def _file_dataset_fn(data_path):
             files_buffer_size).interleave(
                 input_dataset, cycle_length=num_readers, block_length=1)
         # Due to a bug in collection, we sometimes get empty rows.
-        .filter(lambda string: tf.strings.length(string) > 0).apply(
-            tf.data.experimental.shuffle_and_repeat(shuffle_buffer_size)).map(
-                parser_fn, num_parallel_calls=num_map_threads)
+        .filter(lambda string: tf.strings.length(string) > 0)
+        .apply(tf.data.experimental.shuffle_and_repeat(
+                 shuffle_buffer_size,
+                 count=shuffle_repeat_count))
```

Comment:
I don't follow how `shuffle_repeat_count` works in this PR, can you detail a bit (comments, etc)? What accuracy are you measuring?

Reply:
`shuffle_and_repeat` by default has its `count` set to `None`. That means it should repeat forever. We need to set it to 1 to ensure we only iterate over the dataset once rather than getting stuck in an infinite loop. I've added docstrings.